modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java

```diff
@@ -148,4 +149,36 @@ public RelDataType deriveSumType(RelDataTypeFactory typeFactory, RelDataType arg
 
         return typeFactory.createTypeWithNullability(sumType, argumentType.isNullable());
     }
+
+    /**
+     * Checks that {@code toType} and {@code fromType} have compatible type families taking into account custom data types.
+     *
+     * @see SqlTypeUtil#canAssignFrom(RelDataType, RelDataType)
+     */
+    public boolean typeFamiliesAreCompatible(RelDataTypeFactory typeFactory, RelDataType toType, RelDataType fromType) {
```

Comment:
to be honest, I don't understand this method. In my opinion, different real families are never compatible. One exception is for NULL, I guess